eo-phi-normalizer/src/Language/EO/Phi/Rules/Yaml.hs

```diff
@@ -75,9 +75,29 @@ convertRule Rule{..} ctx obj =
   | subst <- matchObject pattern obj
   , all (\cond -> checkCond ctx cond subst) when
   , obj' <- [applySubst subst result]
-  -- TODO #82:20m check that obj' does not have any metavariables
+  , not (objectHasMetavars obj')
   ]
 
+objectHasMetavars :: Object -> Bool
+objectHasMetavars (MetaObject _) = True
+objectHasMetavars (Formation bindings) = any bindingHasMetavars bindings
+objectHasMetavars (Application object bindings) = objectHasMetavars object || any bindingHasMetavars bindings
+objectHasMetavars (ObjectDispatch object attr) = objectHasMetavars object || attrHasMetavars attr
+objectHasMetavars (GlobalDispatch attr) = attrHasMetavars attr
+objectHasMetavars (ThisDispatch attr) = attrHasMetavars attr
+objectHasMetavars _ = False
```

Comment:
I would prefer to not use wildcard here, to ensure we update the code properly if there are any changes to the AST.